core/src/client/mod.rs

```diff
@@ -173,20 +182,21 @@ pub trait TransportReceiverT: 'static {
 }
 
 #[macro_export]
-/// Convert the given values to a [`jsonrpsee_types::ParamsSer`] as expected by a jsonrpsee Client (http or websocket).
+/// Convert the given values to a [`jsonrpsee_types::UnnamedParams`] as expected by a jsonrpsee Client (http or websocket).
 macro_rules! rpc_params {
+	() => {
+		// ToRpcParams is implemented for the empty tuple.
+		()
+	};
 	($($param:expr),*) => {
 		{
-			let mut __params = vec![];
+			let mut __params = $crate::client::__reexports::UnnamedParamsBuilder::new();
 			$(
-				__params.push($crate::client::__reexports::to_json_value($param).expect("json serialization is infallible; qed."));
+				__params.insert($param).expect("json serialization is infallible; qed.");
```

Comment:
Just FYI, Json serialisation isn't infallible?

For example object keys just be strings, so trying to serialize eg a Map<bool, Foo> would fail.

Happy to sort this out in a future PR though to avoid bloating this one; I'd lean towards doing something like I did in subxt and having the macro return a Result, so that the caller can unwrap or "?" It.

Whatcha think @niklasad1 ?

Reply:
yepp, you are correct @jsdw 

as long as we document that it will panic if serialization fails I'm fine with it and remove the incorrect expect :)

it's intended to work as the `serde_json::json! macro` but we can do what you did in subxt here as well.

Reply:
Yep, it makes sense. For this PR have changed the `except` message and added documentation. Will come up with a followup PR to return a `Result` here

Reply:
That makes sense @niklasad1! I pondered this and, indeed, it's only programmer error that can led to the panic (ie providing fallible things to the macro), so I'm ok with keeping the panic behaviour and not returning a result!

Maybe subxt's version should be the same..